src/dsp_tools/commands/xmlupload/iiif_client.py

```diff
@@ -0,0 +1,61 @@
+from dataclasses import dataclass
+from typing import Protocol
+
+import requests
+
+from dsp_tools.commands.xmlupload.models.input_problems import IIIFUriProblem
+
+
+@dataclass(frozen=True)
+class IIIFUriValidator(Protocol):
+    """Interface (protocol) for communication with external IIIF-servers to do a health check."""
+
+    uri: str
+    passed_regex: bool
+
+    def validate(self) -> IIIFUriProblem | None:
+        """Check if the IIIF-server is reachable and if not, it returns information for the user."""
+
+
+@dataclass(frozen=True)
+class IIIFUriValidatorLive:
+    """Client handling communication with external IIIF-servers to do a health check."""
+
+    uri: str
+    passed_regex: bool
+
+    def validate(self) -> IIIFUriProblem | None:
+        """Check if the IIIF-server is reachable and if not it returns information for the user."""
+        response = self._make_network_call()
+        if isinstance(response, Exception):
+            return IIIFUriProblem(uri=self.uri, passed_regex=self.passed_regex, thrown_exception=response)
```

Comment:
I would not pass an exception into a class constructor. From a Software Design point of view, it feels wrong. I think the error message as string does the job.

Reply:
I was undecided about this myself, so I will change it as suggested.